File: Java/Program/Petshop.java
package Java.Program;

import java.util.Scanner;
import java.util.ArrayList;

public class Petshop {
    private String id;
    private String nama;
    private String kategori;
    private int harga;

    public Petshop() {
        this.id = "";
        this.nama = "";
        this.kategori = "";
        this.harga = 0;
    }

    public Petshop(String id, String nama, String kategori, int harga) {
        this.id = id;
        this.nama = nama;
        this.kategori = kategori;
        this.harga = harga;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getNama() {
        return nama;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public String getKategori() {
        return kategori;
    }

    public void setKategori(String kategori) {
        this.kategori = kategori;
    }

    public int getHarga() {
        return harga;
    }

    public void setHarga(int harga) {
        this.harga = harga;
    }

    public void border() {
        System.out.println("+--------------------------------------------+");
    }

    public void add(Scanner sc, ArrayList<Petshop> ls) {
        
        id = sc.next();
        nama = sc.next();
        kategori = sc.next();
        harga = sc.nextInt();
        
        Petshop temp = new Petshop();
        temp.setId(id);
        temp.setNama(nama);
        temp.setKategori(kategori);
        temp.setHarga(harga);

        ls.add(temp);
    }

    public void show(ArrayList<Petshop> ls) {
        System.out.println(" ID | Nama | Kategori | Harga");
        border();
        for (int i = 0; i < ls.size(); i++) {
            System.out.println(ls.get(i).getId() + " " + ls.get(i).getNama() + " " + ls.get(i).getKategori() + " "
                    + ls.get(i).getHarga());
        }
    }

    public Petshop find(Scanner sc, ArrayList<Petshop> ls, String command) {
        String target;
        if (command.equals("find")) {
            System.out.print("input item name: ");
            target = sc.next();
            for (Petshop temp : ls) {
                if (temp.getNama().equals(target)) {
                    return temp;
                }
            }
        } else {
            System.out.print("input item ID: ");
            target = sc.next();
            // proses mencari nama barang dalam list dan "menyimpan alamat" iterator
            for (Petshop temp : ls) {
                if (temp.getId().equals(target)) {
                    return temp;
                }
            }
        }
        border();
        return null;
    }

    public void update(Scanner sc, Petshop it)
    {
        nama = sc.next();
        kategori = sc.next();
        harga = sc.nextInt();
        border();
        // set atribut dengan setter
        it.setNama(nama);
        it.setKategori(kategori);
        it.setHarga(harga);
    }
}

File: PHP/Program/Petshop.php
<?php
class Petshop
{
    // Properties
    private $id, $nama, $kategori, $image;
    private $harga;

    // Constructor
    function __construct()
    {
        $this->id = "";
        $this->nama = "";
        $this->kategori = "";
        $this->harga = 0;
        $this->image = "";
    }
    
    // Setter
    function setID($id) {
        $this->id = $id;
    }
    function setNama($nama) {
        $this->nama = $nama;
    }
    function setKategori($kategori) {
        $this->kategori = $kategori;
    }
    function setHarga($harga) {
        $this->harga = $harga;
    }
    function setImage($image) {
        $this->image = $image;
    }

    // Getter
    function getID() {
        return $this->id;
    }
    function getNama() {
        return $this->nama;
    }
    function getKategori() {
        return $this->kategori;
    }
    function getHarga() {
        return $this->harga;
    }
    function getImage() {
        return $this->image;
    }

    // Method
    function add($id, $nama, $kategori, $harga, $image) {
        $temp = new Petshop();
        $temp->setID($id);
        $temp->setNama($nama);
        $temp->setKategori($kategori);
        $temp->setHarga($harga);
        $temp->setImage($image);
    
        return $temp;
    }
    
    function update($List, $target, $nama, $kategori, $harga, $image) {
        $found = false;
        foreach ($List as $data) {
            if($data->getID() == $target) {
                $data->setNama($nama);
                $data->setKategori($kategori);
                $data->setHarga($harga);
                $data->setImage($image);
                $found = true;
            }
        }
        if($found) {
            echo "item ". $target ." found<br>";
            echo "item with ID ". $target ." updated<br>";
        } else {
            echo "item with ID ". $target ." doesn't exist<br>";
        }
        echo "<br>";
    }
    
    function find($List, $target) {
        foreach ($List as $key=>$data) {
            if($data->getID() == $target) {
                echo "item with ID ". $target ." deleted<br><br>";
                return $key;
            }
        }
        echo "item with ID ". $target ." doesn't exist<br><br>";
        return -1;
    }
    
    function show($List) {
        echo "<table border='1'>";
            echo "<thead>";
                echo "<th>ID</th>";
                echo "<th>Gambar</th>";
                echo "<th>Nama</th>";
                echo "<th>Kategori</th>";
                echo "<th>Harga</th>";
            echo "</thead>";
            
            echo "<tbody>";
                foreach ($List as $data) {
                echo "<tr>";
                    echo "<td>" . $data->getID() . "</td>";
                    echo "<td><img src='img/" . $data->getImage() . "' width='100'/></td>";
                    echo "<td>" . $data->getNama() . "</td>";
                    echo "<td>" . $data->getKategori() . "</td>";
                    echo "<td>" . $data->getHarga() . "</td>";
                echo "</tr>";
                }
            echo "</tbody>";
        echo "</table>";
        echo "<br>";
    }
}
